pkg/workspace/plugins.go

```diff
@@ -598,6 +600,26 @@ func SelectCompatiblePlugin(
 	return bestMatch, nil
 }
 
+// ReadCloserProgressBar displays a progress bar for the given closer and returns a wrapper closer to manipulate it.
+func ReadCloserProgressBar(closer io.ReadCloser, size int64, colorization colors.Colorization) io.ReadCloser {
+	if size == -1 {
+		return closer
+	}
+
+	// If we know the length of the download, show a progress bar.
+	bar := pb.New(int(size))
+	bar.Prefix(colorization.Colorize(colors.SpecUnimportant + "Downloading plugin:"))
```

Comment:
Feels like we should parameterize the message here

Reply:
In what way?

Reply:
https://github.com/pulumi/pulumi/pull/3124/files#r316329605

Reply:
Got you. Agreed! I started with a less generic function name, and didn't think after renaming. Will do.